beacon-chain/rpc/eth/events/events.go

```diff
@@ -62,300 +61,357 @@ var casesHandled = map[string]bool{
-				log.WithError(err).Error("Unable to obtain stream payload attributes")
-			}
-			return nil
+			s.sendPayloadAttributes(ctx, w, flusher)
 		}
-		return nil
 	case statefeed.FinalizedCheckpoint:
 		if _, ok := requestedTopics[FinalizedCheckpointTopic]; !ok {
-			return nil
+			return
 		}
-		finalizedCheckpoint, ok := event.Data.(*ethpb.EventFinalizedCheckpoint)
+		checkpointData, ok := event.Data.(*ethpb.EventFinalizedCheckpoint)
 		if !ok {
-			return nil
+			write(w, flusher, topicDataMismatch, event.Data, FinalizedCheckpointTopic)
+			return
+		}
+		checkpoint := &FinalizedCheckpointEvent{
+			Block:               hexutil.Encode(checkpointData.Block),
+			State:               hexutil.Encode(checkpointData.State),
+			Epoch:               fmt.Sprintf("%d", checkpointData.Epoch),
+			ExecutionOptimistic: checkpointData.ExecutionOptimistic,
 		}
-		return streamData(stream, FinalizedCheckpointTopic, finalizedCheckpoint)
+		send(w, flusher, FinalizedCheckpointTopic, checkpoint)
 	case statefeed.Reorg:
 		if _, ok := requestedTopics[ChainReorgTopic]; !ok {
-			return nil
+			return
 		}
-		reorg, ok := event.Data.(*ethpb.EventChainReorg)
+		reorgData, ok := event.Data.(*ethpb.EventChainReorg)
 		if !ok {
-			return nil
+			write(w, flusher, topicDataMismatch, event.Data, ChainReorgTopic)
+			return
+		}
+		reorg := &ChainReorgEvent{
+			Slot:                fmt.Sprintf("%d", reorgData.Slot),
+			Depth:               fmt.Sprintf("%d", reorgData.Depth),
+			OldHeadBlock:        hexutil.Encode(reorgData.OldHeadBlock),
+			NewHeadBlock:        hexutil.Encode(reorgData.NewHeadBlock),
+			OldHeadState:        hexutil.Encode(reorgData.OldHeadState),
+			NewHeadState:        hexutil.Encode(reorgData.NewHeadState),
+			Epoch:               fmt.Sprintf("%d", reorgData.Epoch),
+			ExecutionOptimistic: reorgData.ExecutionOptimistic,
 		}
-		return streamData(stream, ChainReorgTopic, reorg)
+		send(w, flusher, ChainReorgTopic, reorg)
 	case statefeed.BlockProcessed:
 		if _, ok := requestedTopics[BlockTopic]; !ok {
-			return nil
+			return
 		}
 		blkData, ok := event.Data.(*statefeed.BlockProcessedData)
 		if !ok {
-			return nil
-		}
-		v1Data, err := migration.BlockIfaceToV1BlockHeader(blkData.SignedBlock)
-		if err != nil {
-			return err
+			write(w, flusher, topicDataMismatch, event.Data, BlockTopic)
+			return
 		}
-		item, err := v1Data.Message.HashTreeRoot()
+		blockRoot, err := blkData.SignedBlock.Block().HashTreeRoot()
 		if err != nil {
-			return errors.Wrap(err, "could not hash tree root block")
+			write(w, flusher, "Could not get block root: "+err.Error())
+			return
 		}
-		eventBlock := &ethpb.EventBlock{
-			Slot:                blkData.Slot,
-			Block:               item[:],
+		blk := &BlockEvent{
+			Slot:                fmt.Sprintf("%d", blkData.Slot),
+			Block:               hexutil.Encode(blockRoot[:]),
 			ExecutionOptimistic: blkData.Optimistic,
 		}
-		return streamData(stream, BlockTopic, eventBlock)
-	default:
```

Comment:
Should we not keep a state default and have it log something rather than just `return nil` if we have an unexpected state ? Or does this just never happen?

Reply:
If an event that we don't intend to handle is emitted, we should just ignore it. No need to log e.g. "Foo event observed" when there is nothing we can (nor want to) do about it.